src/plugins/gimbal/gimbal_protocol_v1.cpp

```diff
@@ -156,4 +156,56 @@ void GimbalProtocolV1::set_roi_location_async(
         });
 }
 
+Gimbal::Result GimbalProtocolV1::take_control(Gimbal::ControlMode control_mode)
+{
+    _current_control_status.control_mode = control_mode;
+    return Gimbal::Result::Success;
+}
+
+void GimbalProtocolV1::take_control_async(
+    Gimbal::ControlMode control_mode, Gimbal::ResultCallback callback)
+{
+    _current_control_status.control_mode = control_mode;
+
+    if (callback) {
+        _system_impl.call_user_callback([callback]() { callback(Gimbal::Result::Success); });
+    }
+}
+
+Gimbal::Result GimbalProtocolV1::release_control()
+{
+    _current_control_status.control_mode = Gimbal::ControlMode::None;
+    return Gimbal::Result::Success;
+}
+
+void GimbalProtocolV1::release_control_async(Gimbal::ResultCallback callback)
+{
+    _current_control_status.control_mode = Gimbal::ControlMode::None;
+
+    if (callback) {
+        _system_impl.call_user_callback([callback]() { callback(Gimbal::Result::Success); });
+    }
+}
+
+Gimbal::ControlStatus GimbalProtocolV1::control()
+{
+    return _current_control_status;
+}
+
+void GimbalProtocolV1::control_async(Gimbal::ControlCallback callback)
+{
+    _control_callback = callback;
+    _control_thread_cv.notify_one();
+
+    std::thread([this, &callback]() {
+        std::unique_lock<std::mutex> lock(_control_thread_mutex);
+
+        while (_control_callback) {
+            _system_impl.call_user_callback(
+                [this, callback]() { callback(_current_control_status); });
+            _control_thread_cv.wait_for(lock, std::chrono::seconds(1));
+        }
+    });
+}
```

Comment:
I don't really get that. I assume it's overkill that we could just ignore. I guess we'll leave it in now and clean it up later.

Reply:
I just want to allow an application to use the same logic for both v1 and v2. So you need to get some status updates if you expect them, right? :blush:

Reply:
Hm, fair enough. If it works you can leave it in :)